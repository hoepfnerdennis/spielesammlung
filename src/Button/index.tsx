import React from 'react';
import { useFela } from 'react-fela';
import { IStyle } from 'fela';

const buttonStyle = (secondary?: boolean) => (): IStyle => ({
  height: '2rem',
  fontSize: '1rem',
  border: '2px solid darkcyan',
  borderRadius: '5px',
  padding: '0 1rem',
  whiteSpace: 'nowrap',
  backgroundColor: secondary ? 'white' : 'darkcyan',
  color: secondary ? 'darkcyan' : 'white',
});

const Button: React.SFC<React.PropsWithChildren<{ secondary?: boolean; onClick: () => void }>> = ({
  children,
  secondary,
  onClick,
}): JSX.Element => {
  const { css } = useFela();
  return (
    <button type="button" className={css(buttonStyle(secondary))} onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
