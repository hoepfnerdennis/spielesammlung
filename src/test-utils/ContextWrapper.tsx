import React, { FunctionComponent } from 'react';
import { createRenderer } from 'fela';
import { RendererProvider } from 'react-fela';

const renderer = createRenderer();

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const ContextWrapper: FunctionComponent = ({ children }) => {
  return <RendererProvider renderer={renderer}>{children}</RendererProvider>;
};

export default ContextWrapper;
