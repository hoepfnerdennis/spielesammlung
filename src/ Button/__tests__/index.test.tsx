import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../index';
import ContextWrapper from '../../test-utils/ContextWrapper';

describe('Button', () => {
  it.each([
    ['default', false],
    ['secondary', true],
  ])('should render %s', (desc, secondary) => {
    const { container } = render(
      <ContextWrapper>
        <Button onClick={jest.fn()} secondary={secondary}>
          Button
        </Button>
      </ContextWrapper>
    );
    expect(container).toMatchSnapshot();
  });
  it('should handle onClikc', () => {
    const onClick = jest.fn();
    const { getByText } = render(
      <ContextWrapper>
        <Button onClick={onClick}>Button</Button>
      </ContextWrapper>
    );
    fireEvent.click(getByText('Button'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
