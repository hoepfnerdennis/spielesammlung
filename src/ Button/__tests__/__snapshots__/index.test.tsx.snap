// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render default 1`] = `
<div>
  <button
    class="filled"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`Button should render secondary 1`] = `
<div>
  <button
    class="stroke"
    type="button"
  >
    Button
  </button>
</div>
`;
